File: mango-genome/backend/schemas/msa.py
from typing import List, Optional
from pydantic import BaseModel


class AlignmentStats(BaseModel):
    alignment_length: int
    average_identity: float
    average_gaps: float
    total_sequences: int
    pairwise_identities: List[float]


class DotPlot(BaseModel):
    label: str
    image: str  # base64 PNG


class MsaPlots(BaseModel):
    identity_heatmap: str        # base64 PNG
    base_frequency: str          # base64 PNG
    pairwise_distribution: str   # base64 PNG
    dot_plots: List[DotPlot]


class MsaResponse(BaseModel):
    aligned_fasta: str
    formatted_alignment: str
    stats: AlignmentStats
    plots: Optional[MsaPlots] = None
    report_text: str

File: mango-genome/frontend/src/types/msa.ts
export interface AlignmentStats {
  alignment_length: number;
  average_identity: number;
  average_gaps: number;
  total_sequences: number;
  pairwise_identities: number[];
}

export interface DotPlot {
  label: string;
  image: string; // base64 PNG
}

export interface MsaPlots {
  identity_heatmap: string;
  base_frequency: string;
  pairwise_distribution: string;
  dot_plots: DotPlot[];
}

export interface MsaResponse {
  aligned_fasta: string;
  formatted_alignment: string;
  stats: AlignmentStats;
  plots: MsaPlots | null;
  report_text: string;
}
